src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -0,0 +1,211 @@
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.models.TrackedJourney;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.testutils.CommonTestUtils;
+import org.opentripplanner.middleware.testutils.OtpMiddlewareTestEnvironment;
+import org.opentripplanner.middleware.triptracker.ManageLegTraversal;
+import org.opentripplanner.middleware.triptracker.ManageTripTracking;
+import org.opentripplanner.middleware.triptracker.TrackingLocation;
+import org.opentripplanner.middleware.triptracker.TripStatus;
+import org.opentripplanner.middleware.utils.Coordinates;
+import org.opentripplanner.middleware.utils.JsonUtils;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.opentripplanner.middleware.auth.Auth0Connection.restoreDefaultAuthDisabled;
+import static org.opentripplanner.middleware.auth.Auth0Connection.setAuthDisabled;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.getTimeInMilliseconds;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.interpolatePoints;
+
+public class ManageLegTraversalTest extends OtpMiddlewareTestEnvironment {
+
+    private static MonitoredTrip monitoredTrip;
+
+    private static TrackedJourney trackedJourney;
+
+    private static final int SUBWAY_LEG_ID = 1;
+
+    private static final int FINAL_LEG_ID = 2;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        assumeTrue(IS_END_TO_END);
+        setAuthDisabled(false);
+        String payload = CommonTestUtils.getTestResourceAsString("controllers/api/monitored-trip-for-tracking.json");
+        monitoredTrip = JsonUtils.getPOJOFromJSON(payload, MonitoredTrip.class);
+        Persistence.monitoredTrips.create(monitoredTrip);
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        restoreDefaultAuthDisabled();
+        monitoredTrip = Persistence.monitoredTrips.getById(monitoredTrip.id);
+        if (monitoredTrip != null) monitoredTrip.delete();
+    }
+
+    @AfterEach
+    public void tearDownAfterTest() {
+        if (trackedJourney != null) {
+            trackedJourney.delete();
+            trackedJourney = null;
+        }
+    }
+
+    // This is WIP. These real points and time are nowhere near the expected. This is probably due to the speed of the
+    // subway.
```

Comment:
We should expect disruption in GPS signal from travel through certain tunnels and I think that is what is happening in the sample.

Reply:
I've plotted the first 100 lat/lon points. The first is the trip, the second the provided traveller positions. The data isn't accurate enough to associate the two.